### PR TITLE
vs eng audio settings functionality lv 412

### PR DESCRIPTION
**Pull request by Titonfish**

### CODE REVIEW

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+**********************************************************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using FMOD.Studio;
+using System.IO;
+
+public class AudioSettings : MonoBehaviour
```
Comment (ASLH68) [inline Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:16]: XML comment

**Review by ASLH68 — APPROVED** [review]

**Review by AlecatEnjoyer — APPROVED** [review]
I approve

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+{
+    private VCA _vca;
+
+    [SerializeField] private string _audioSettingsFilePath;
+
+    [SerializeField] private Slider _masterSlider;
```
Comment (AlecatEnjoyer) [inline Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:22]: might be best to put tooltips here just incase someone miss interprets something.

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using FMOD.Studio;
+using System.IO;
+
+public class AudioSettings : MonoBehaviour
+{
+    private VCA _vca;
```
Comment (RyanASwanson) [inline Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:18]: This name is very unclear

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+            }
+        }
+
+        return volumeArray;
+    }
+
+    /// <summary>
+    /// Save the volumes to the save file
+    /// </summary>
+    private void SaveData()
```
Comment (RyanASwanson) [inline Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:160]: I feel this functionality could be done through our Save Manager, but I'm not going to request changes as it's a bit late to be doing so

**Review by RyanASwanson — APPROVED** [review]
Approved with comments

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs (hunk 4 above)
Comment (Titonfish) [reply Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:160]: I wasn't even aware we had save functionality until after this was written, I can fix it next semester

File: Assets/General/Scripts/MenuFunctionality/AudioSettings.cs (hunk 3 above)
Comment (Titonfish) [reply Assets/General/Scripts/MenuFunctionality/AudioSettings.cs:18]: dude I wish I knew what this was too tbh

**Pull request merged**